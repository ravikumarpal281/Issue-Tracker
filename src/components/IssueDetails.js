import axios from "axios";
import React, { useEffect, useState } from "react";
import { Col, Container, Row, Button, Card } from "react-bootstrap";
import { Link, withRouter } from "react-router-dom";

const IssueDetails = (props) => {
  console.log("Issue Details PROPS", props);
  const [state, setstate] = useState({
    id: "",
    name: "",
    description: "",
    severity: "",
    status: "",
  });

  const handleDeleteIssue = () => {
    if (window.confirm("Are you sure you want to delete the issue?")) {
      axios
        .delete(`http://localhost:30001/Issues/${props.id.params.id}`)
        .catch((error) => console.error(error.message));
      props.history.push({ pathname: "/" });
    }
  };

  const handleEditIssue = () => {
    if (window.confirm("Are you sure you want to edit the issue?")) {
      console.log("Edit Issue Click", props.id.params.id);
      props.history.push({ pathname: "/editIssue/" + props.id.params.id });
    }
  };

  useEffect(() => {
    axios
      .get(`http://localhost:30001/Issues/${props.id.params.id}`)
      .then((response) => setstate(response.data))
      .catch((error) => console.error(error.message));
  }, []);

  return (
    <div>
      <Container>
        <h1>Issue Details</h1>
        <Card className="sign-in-form">
          <Row>
            <Col sm={8} className="form-style">
              <p>
                <strong>Name : </strong> {state.name}
              </p>
              <p>
                <strong>Issue Description : </strong> {state.description}
              </p>
              <p>
                <strong>Severity : </strong> {state.severity}
              </p>
              <p>
                <strong>Status : </strong> {state.status}
              </p>
              <Button variant="outline-primary" onClick={handleEditIssue}>
                Edit Issue
              </Button>{" "}
              <Button variant="outline-danger" onClick={handleDeleteIssue}>
                Delete Issue
              </Button>{" "}
            </Col>
          </Row>
        </Card>
      </Container>
    </div>
  );
};

export default withRouter(IssueDetails);
