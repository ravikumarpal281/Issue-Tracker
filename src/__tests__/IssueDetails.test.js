import React from "react";
import ReactDOM from "react-dom";
import toJson from "enzyme-to-json";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import IssueDetails from "../components/IssueDetails";
import { Spinner, Button } from "react-bootstrap";

Enzyme.configure({ adapter: new Adapter() });

const props = {
  history: {},
  id: {
    path: "/issue/:id",
    url: "/issue/1",
    isExact: true,
    params: {
      id: "1",
    },
  },
  location: {},
  match: {},
};
it("Issue Details Snap", () => {
  const wrapper = shallow(<IssueDetails.WrappedComponent id={props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("Issue Details Headers Test", () => {
  const wrapper = shallow(<IssueDetails.WrappedComponent id={props} />);
  expect(wrapper.find("div").length).toEqual(1);
  expect(wrapper.find("h1").text()).toEqual("Issue Details");
  expect(wrapper.find("p").length).toEqual(4);
  expect(wrapper.find(Button).length).toEqual(2);
});

it("Issue Details Labels Test", () => {
  const wrapper = shallow(<IssueDetails.WrappedComponent id={props} />);
  expect(wrapper.find("p").length).toEqual(4);
});

it("Edit And Delete Button Check", () => {
  const wrapper = shallow(<IssueDetails.WrappedComponent id={props} />);
  expect(wrapper.find(Button).length).toEqual(2);
});

it("Edit Button Check", () => {
  const mockEdit = jest.fn();
  const wrapper = shallow(<Button onClick={mockEdit}>Edit Issue</Button>);
  const editButton = wrapper.find({ children: "Edit Issue" });
  expect(editButton).toBeTruthy();
  editButton.simulate("click");
  expect(mockEdit).toHaveBeenCalled();
});

it("Delete Button Check", () => {
  const mockEdit = jest.fn();
  const wrapper = shallow(<Button onClick={mockEdit}>Delete Issue</Button>);
  const editButton = wrapper.find({ children: "Delete Issue" });
  expect(editButton).toBeTruthy();
  editButton.simulate("click");
  expect(mockEdit).toHaveBeenCalled();
});
