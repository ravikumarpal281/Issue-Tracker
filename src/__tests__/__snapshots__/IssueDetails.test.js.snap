// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue Details Snap 1`] = `
<div>
  <Container
    fluid={false}
  >
    <h1>
      Issue Details
    </h1>
    <Card
      body={false}
      className="sign-in-form"
    >
      <Row
        noGutters={false}
      >
        <Col
          className="form-style"
          sm={8}
        >
          <p>
            <strong>
              Name : 
            </strong>
             
          </p>
          <p>
            <strong>
              Issue Description : 
            </strong>
             
          </p>
          <p>
            <strong>
              Severity : 
            </strong>
             
          </p>
          <p>
            <strong>
              Status : 
            </strong>
             
          </p>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="outline-primary"
          >
            Edit Issue
          </Button>
           
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="outline-danger"
          >
            Delete Issue
          </Button>
           
        </Col>
      </Row>
    </Card>
  </Container>
</div>
`;
